lab/skel_graph/traversal/problems/muzeu/main.cpp

```diff
@@ -0,0 +1,132 @@
+// Copyright Radu Nichita 2020 radunichita99@gmail.com
+
+// Muzeu - Problema parcurgeri
+// BFS cu mai multe surse
+
+// Complexitate O(N ^ 2)
+// N este dimensiunea matricei
+
+// https://infoarena.ro/problema/muzeu
+
+#include <bits/stdc++.h>
+#define KMAX 300
+#define kInf (1 << 30)
+
+class Task {
+    // coada cu nodurile ce urmeaza a fi explorate
+    std::queue<std::pair<int, int>> explore;
```

Comment:
Please keep convension presented in the bfs source:

- `queue -> q` and declared in a bfs function (you should declare a variable only in the scope where is needed, otherwise you can move all as global variables)

- same for `distances` and `directions`.

The idea is to use this repo as materials for students. We cannot have n conventions for the same basic algorithm.

Reply:
I think q can't be declared in the bfs function, as it is used in the read input to put the cops from which bfs will actually start.